src/main/scala/com/fulcrumgenomics/util/IntervalListSource.scala

```diff
@@ -0,0 +1,126 @@
+
+import com.fulcrumgenomics.FgBioDef.PathToIntervals
+import com.fulcrumgenomics.commons.CommonsDef.BetterBufferedIteratorScalaWrapper
+import com.fulcrumgenomics.commons.util.StringUtil
+import htsjdk.samtools.util.{BufferedLineReader, Interval}
+import htsjdk.samtools.{SAMFileHeader, SAMSequenceDictionary, SAMTextHeaderCodec}
+
+import scala.io.Source
+
+object IntervalListSource {
+
+  /** Creates a new interval list source from a sequence of lines. */
+  def apply(lines: Seq[String]): IntervalListSource = new IntervalListSource(lines.iterator)
+
+  /** Creates a new interval list source from an iterator of lines. */
+  def apply(lines: Iterator[String]): IntervalListSource = new IntervalListSource(lines)
+
+  /** Creates a new interval list source from an input stream. */
+  def apply(stream: InputStream): IntervalListSource = new IntervalListSource(Source.fromInputStream(stream).getLines(), Some(stream))
+
+  /** Creates a new interval list source from a source. */
+  def apply(source: Source): IntervalListSource = new IntervalListSource(source.getLines(), Some(source))
+
+  /** Creates a new interval list source from a File. */
+  def apply(file: File): IntervalListSource = apply(path=file.toPath)
+
+  /** Creates a new interval list source from a Path. */
+  def apply(path: PathToIntervals): IntervalListSource = apply(Io.toInputStream(path))
+}
+
+/**
+  * Reads intervals from any text based source via a reader.  The underlying reader is closed automatically when EOF is
+  * reached.
+  */
+class IntervalListSource private(lines: Iterator[String],
+                                 private[this] val source: Option[{ def close(): Unit }] = None)
+  extends Iterator[Interval] with Closeable {
+
+  private val iter = lines.bufferBetter
+
+  private var lineNumber = 1L
+
+  // Read the header
+  val header: SAMFileHeader = {
+    val codec = new SAMTextHeaderCodec
+    val headerLines = iter.takeWhile(_.startsWith("@")).toSeq
+    require(headerLines.nonEmpty, "No header found")
+    lineNumber += headerLines.length
+    val lineReader = BufferedLineReader.fromString(headerLines.mkString("\n"))
+    codec.decode(lineReader, "IntervalListSource")
+  }
+
+  require(!this.dict.isEmpty, "No reference sequences found in the header.")
+
+  def dict: SAMSequenceDictionary= this.header.getSequenceDictionary
+
+  private val parseArray = Array[String]("", "", "", "", "")
+
+  /** True if calling `next()` will yield another record, false otherwise. */
+  override def hasNext: Boolean = this.iter.nonEmpty
+
+  /** Returns the next record if available, or throws an exception if none is available. */
+  override def next(): Interval = parse(this.iter.next())
+
+  /** Closes the underlying reader; only necessary if EOF hasn't been reached. */
+  override def close(): Unit = this.source.foreach(_.close())
+
+  private def parse(line: String): Interval = {
+    require(StringUtil.split(line, '\t', parseArray) == 5, s"Expected 5 fields on line $lineNumber")
```

Comment:
Could you split this into two lines please?  I.e.
```scala
val fieldCount = StringUtil.split(line, '\t', parseArray)
require(fieldCount == 5, "...")
```
I don't like having side-effecting things in `require()`s